src/components/tabs/js/tabsController.js

```diff
@@ -169,6 +170,8 @@ function MdTabsController ($scope, $element, $window, $mdConstant, $mdTabInkRipp
 
   function handleMaxTabWidth (newWidth, oldWidth) {
     if (newWidth !== oldWidth) {
+      var elements = getElements();
```

Comment:
`getElements` queries the DOM, right? I _think_ (not really sure) it could be avoided by using the data from the mutation observer, which should return a list of the added/removed nodes, which can be used to update the references in the list.